pkg/api/testapi/testapi.go

```diff
@@ -200,3 +200,7 @@ func GetCodecForObject(obj runtime.Object) (runtime.Codec, error) {
 	}
 	return nil, fmt.Errorf("unexpected kind: %v", kind)
 }
+
+func NewTestGroup(external, internal unversioned.GroupVersion) TestGroup {
+	return TestGroup{external, internal}
+}
```

Comment:
D'oh. You're right. Fine :)

On Fri, Dec 11, 2015 at 11:10 AM, Chao Xu <notifications@github.com> wrote:

> In pkg/api/testapi/testapi.go
> <https://github.com/kubernetes/kubernetes/pull/18231#discussion_r47391902>
> :
>
> > @@ -200,3 +200,7 @@ func GetCodecForObject(obj runtime.Object) (runtime.Codec, error) {
> >  	}
> >  	return nil, fmt.Errorf("unexpected kind: %v", kind)
> >  }
> > +
> > +func NewTestGroup(external, internal unversioned.GroupVersion) TestGroup {
> > +	return TestGroup{external, internal}
> > +}
>
> The two fields of TestGroup are private. When I construct the TestGroup
> using struct literals in my test the compiler complains.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18231/files#r47391902>.
>